alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/alerts/AlertMover.kt

```diff
@@ -94,15 +93,15 @@ class AlertMover(
 
     private fun onSearchResponse(response: SearchResponse) {
         // If no alerts are found, simply return
-        if (response.hits.totalHits == 0L) return
+        if (response.hits.totalHits.value == 0L) return
         val indexRequests = response.hits.map { hit ->
-            IndexRequest(AlertIndices.HISTORY_WRITE_INDEX, AlertIndices.MAPPING_TYPE)
+            IndexRequest(AlertIndices.HISTORY_WRITE_INDEX)
                     .routing(monitorId)
                     .source(Alert.parse(alertContentParser(hit.sourceRef), hit.id, hit.version)
                             .copy(state = Alert.State.DELETED)
                             .toXContent(XContentFactory.jsonBuilder(), ToXContent.EMPTY_PARAMS))
-                    .version(hit.version)
-                    .versionType(VersionType.EXTERNAL_GTE)
+                    .setIfSeqNo(hit.seqNo)
```

Comment:
https://github.com/opendistro-for-elasticsearch/alerting/blob/10d51efa2c1dc15e8c9ac0d33e8566aa2fa24fdd/alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/alerts/AlertMover.kt#L98-L104

This doesn't seem correct - how will the `seqNo` and `primaryTerm` be retained when moving from the active index to the history index?

Reply:
I am not sure what you are asking. We were keeping the version prior, which is still supported since we are using an external versioning system, but since versions are going away, we might want to keep the document the same as a direct copy from the `ACTIVE` state to `DELETED` state between the indexes.

Reply:
Sequence numbers and primary terms are specific to an index. They cannot be carried over from the `hit` in the active index to the copy in the history index.  To avoid stale writes on the alert in the history index we need to continue to use external versioning which is still supported in ES 7: https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking-changes-7.0.html#_internal_versioning_is_no_longer_supported_for_optimistic_concurrency_control